veloroutes_voies_vertes/processing/covadis/export_covadis.py

```diff
@@ -145,10 +145,23 @@ def createExportTable(table, connection):
             msg = e.args[0]
             raise QgsProcessingException(msg)
 
-    def to_shapefile(self, context, table, dpt, dirname, uri, geom, charger):
-        uri.setDataSource('exports', table, geom, "")
-        layer = QgsVectorLayer(uri.uri(), table, "postgres")
+    def to_shapefile(self, context, table, dpt, dirname, uri, geom, charger, feedback):
+        plugin_dir = plugin_path()
+        sql_file = os.path.join(plugin_dir, "install/sql/export/export_{}.sql".format(table))
+        sql = ""
+        with open(sql_file, "r") as f:
+            sql = f.read()
+            if len(sql.strip()) == 0:
+                feedback.pushInfo("Fichier vide")
+        feedback.pushInfo(sql)
+        uri.setDataSource('', '(' + sql + ')', geom, '', 'id_iti')
+        layer = QgsVectorLayer(uri.uri(False), table, "postgres")
+        lay_fields = layer.fields()
+        for fi in lay_fields:
+            feedback.pushInfo("######")
+            feedback.pushInfo(fi.name())
         if not layer.isValid():
+            feedback.pushInfo("Layer not valid " + layer.name())
             return False
```

Comment:
Si la méthode `to_shapefile` renvoit un `QgsVectorLayer`, je recommanderais de renvoyer Null ou None plutôt qu'un booléen

Reply:
Work in progress